src/dune_rules/cram_stanza.ml

```diff
@@ -36,5 +37,23 @@ let decode =
        field "applies_to" decode_applies_to ~default:default_applies_to
      and+ alias = field_o "alias" Alias.Name.decode
      and+ deps = field_o "deps" (Bindings.decode Dep_conf.decode)
-     and+ enabled_if = Enabled_if.decode ~allowed_vars:Any ~since:None () in
-     { loc; alias; deps; enabled_if; applies_to })
+     and+ enabled_if = Enabled_if.decode ~allowed_vars:Any ~since:None ()
+     and+ project = Dune_project.get_exn ()
+     and+ package_name =
+       field_o "package"
+         ( Dune_lang.Syntax.since Stanza.syntax (2, 8)
+         >>> Package.Name.decode_loc )
+     in
+     let package =
+       let open Option.O in
+       let+ loc, name = package_name in
+       let packages = Dune_project.packages project in
+       match Package.Name.Map.find packages name with
+       | Some pkg -> (loc, pkg)
+       | None ->
+         User_error.raise ~loc
+           [ Pp.textf "The current scope doesn't define package %S"
+               (Package.Name.to_string name)
+           ]
```

Comment:
Why not use `Stanza_common.Pkg.field`?

Reply:
`Stanza_common.Pkg.field` has a defaulting behavior that doesn't work for our case. If the package field isn't specified by the user, `Pkg.field` will try to guess it and fail if it cannot do so unambiguously. It also has a confusing error message for this use case.

Let me try factoring out the two functions however.

Reply:
Done